src/App.jsx

```diff
@@ -1,28 +1,102 @@
+      },
+      {
+        id: 'njskvbl095',
+        title: 'Title 5',
+        text: 'It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passage.',
+      },
+      {
+        id: 'ibjkhbkjs985',
+        title: 'Title 6',
+        text: 'The standard chunk of Lorem Ipsum used since the 1500s is reproduced below for those interested. Sections 1.10.32 and 1.10.33 from "de Finibus Bonorum et Malorum" by Cicero are also reproduced in their exact original form, accompanied by English versions from the 1914 translation by H. Rackham.',
+      },
+      {
+        id: 'bvdbsjvb34657',
+        title: 'Title 7',
+        text: 'It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.',
+      },
+      {
+        id: 'vsjhvbh354',
+        title: 'Title 8',
+        text: 'Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..", comes from a line in section 1.10.32.',
+      },
+      {
+        id: 'suisfid985',
+        title: 'Title 9',
+        text: 'Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for lorem ipsum will uncover many web sites still in their infancy.',
+      },
+      {
+        id: 'vcbxcvxzse328',
+        title: 'Title 10',
+        text: 'by injected humour, or randomised words which look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary',
+      },
+    ],
+    viewOnlyHandler: false,
   };
 
   // Add new text to stage
   newTextChangeHandler = (value) => {
-    this.setState({ ...value });
+    const itemIndex = this.state.cards.findIndex((item => {
+      return item.id === value.id
+    }));
+    let card = { ...this.state.cards[itemIndex] };
+    card = { ...value };
+    const newCardState = [...this.state.cards];
+    newCardState[itemIndex] = card;
+    this.setState({ cards: newCardState });
+  }
+
+  newViewChangeHandler = () => {
+    this.setState({ viewOnlyHandler: !this.state.viewOnlyHandler })
   }
 
   render() {
+    const tasks = this.state.cards.map((card) => {
+      return (
+        <Task
+          key={card.id}
+          task={card}
+          view={this.state.viewOnlyHandler}
+          newTextChangeHandler={this.newTextChangeHandler}
```

Comment:
When passing an event handler - there is an approach to add 'on' prefix to its name, so in children component we could know which of passed props is a handler to subscribe onto.

Reply:
Fixed